packages/core/src/browser/preferences/preference-service.ts

```diff
@@ -476,6 +487,50 @@ export class PreferenceServiceImpl implements PreferenceService {
         return value;
     }
 
+    protected getScopedValueFromInspection<T>(inspection: PreferenceInspection<T>, scope: PreferenceScope): T | undefined {
+        switch (scope) {
+            case PreferenceScope.Default:
+                return inspection.defaultValue;
+            case PreferenceScope.User:
+                return inspection.globalValue;
+            case PreferenceScope.Workspace:
+                return inspection.workspaceValue;
+            case PreferenceScope.Folder:
+                return inspection.workspaceFolderValue;
+            default: // Should be unreachable
+                return undefined;
```

Comment:
FYI: If you want to handle the _"should be unreachable"_ part the TypeScript way, you have to use `never`, and you will get a compile-time error if you add a new enum:
```ts
    protected getScopedValueFromInspection<T>(inspection: PreferenceInspection<T>, scope: PreferenceScope): T | undefined {
        const handleUnexpectedScope: (never: never) => never = () => {
            throw new Error('Should be unreachable');
        };
        switch (scope) {
            case PreferenceScope.Default:
                return inspection.defaultValue;
            case PreferenceScope.User:
                return inspection.globalValue;
            case PreferenceScope.Workspace:
                return inspection.workspaceValue;
            case PreferenceScope.Folder:
                return inspection.workspaceFolderValue;
            default: // Should be unreachable
                return handleUnexpectedScope(scope);
        }
    }
```
You do not have to change anything, though 😊 I just wanted to bring this up; please ignore if you already know this.

Reply:
Very nice - good to make the person who changes the enum pay 😅